tests/code/test_users.py

```diff
@@ -34,63 +33,78 @@ def test_isEligibleForProgram():
     eligible = isEligibleForProgram(4, "ayisie")
     assert eligible
 
-    # user that is banned from a program
-    eligible = isEligibleForProgram(3,  "khatts")
-    assert not eligible
-
-    # user hasn't attend the required event
-    eligible3 = isEligibleForProgram(1, "ayisie")
-    assert not eligible3
 
 @pytest.mark.integration
-def test_addRemoveInterest():
-    user = "ramsayb2"
-    program_id = 1
+def test_addUserInterest():
+    username = "ramsayb2"
+    program_id = 2
 
-    # test adding and removing test cases for different users
-    rule = 'addInterest'
-    result = addRemoveInterest(rule, program_id, user)
+    # test adding interest for different users
+    result = addUserInterest(program_id, username)
     assert result == "Successfully added interest"
 
-    rule = 'deleteInterest'
-    result = addRemoveInterest(rule, program_id, user)
-    assert result == 'Successfully removed interest'
 
-    rule = 'addInterest'
-    user = "khatts"
-    result = addRemoveInterest(rule, program_id, user)
+    username = "khatts"
+    result = addUserInterest(program_id, username)
     assert result == 'Successfully added interest'
 
-    # test add and removing interest with different program id
-    rule = 'addInterest'
+    # test adding interest with different program id
     program_id = 3
-    result = addRemoveInterest(rule, program_id, user)
+    result = addUserInterest(program_id, username)
     assert result == 'Successfully added interest'
 
-    rule = 'deleteInterest'
-    result = addRemoveInterest(rule, program_id, user)
-    assert result == 'Successfully removed interest'
+
 
 @pytest.mark.integration
-def test_addRemoveInvalidInterest():
+def test_removeUserInterestt():
+
+    #test for removing interest
+    username = "ramsayb2"
+    program_id = 2
+    result = removeUserInterest(program_id, username)
+    assert result == "Successfully removed interest"
+
+    username = "khatts"
+    program_id = 2
+    result = removeUserInterest(program_id, username)
+    assert result == "Successfully removed interest"
+
+    #test removing interest with different program id
+    program_id = 3
+    result = removeUserInterest(program_id, username)
+    assert result == "Successfully removed interest"
```

Comment:
You are only checking successes. No failure cases.

Reply:
Fixed!